spec/integration/puck_spec.rb

```diff
@@ -5,25 +5,31 @@
 
 
 describe 'bin/puck' do
-  let :jar_command do
-    %(GEM_HOME='' GEM_PATH='' java -jar spec/resources/example_app/build/example_app.jar)
+  APP_DIR = File.expand_path('../../resources/example_app', __FILE__)
+
+  def isolated_run(cmd)
+    Dir.chdir(APP_DIR) do
+      `#{cmd}`
+    end
+  end
+
+  def jar_command(cmd)
+    sprintf('java -jar build/example_app.jar %s 2>&1', cmd)
   end
 
   before :all do
-    system([
-      'cd spec/resources/example_app',
-      'rm -rf build',
-      'BUNDLE_WITHOUT=not_installed BUNDLE_GEMFILE=$(pwd)/Gemfile rvm ${RUBY_VERSION}@$(cat .ruby-gemset) do bundle exec puck --extra-files config/app.yml',
-    ].join(' && '))
+    isolated_run("rm -rf build && BUNDLE_GEMFILE=Gemfile BUNDLE_PATH=../../../vendor/example_app-bundle/jruby/1.9 BUNDLE_WITHOUT=not_installed .bundle/bin/puck --extra-files config/app.yml")
```

Comment:
`rm -rf` actually serves multiple purposes here :) JRuby doesn't launch a proper shell for simple commands and will try to run `BUNDLE_GEMFILE` if removed. This could however be achieved by including any special character somewhere in the command, e.g. `&`, `|`, `>`, etc.